src/Host/Client/Impl/Host/RemoteCredentialsDecorator.cs

```diff
@@ -38,31 +32,22 @@ internal class RemoteCredentialsDecorator : ICredentialsDecorator {
             }
         }
 
-        public async Task<IDisposable> LockCredentialsAsync(CancellationToken cancellationToken = default(CancellationToken)) {
+        public async Task LockCredentialsAsync(CancellationToken cancellationToken = default(CancellationToken)) {
```

Comment:
This method doesn't lock anything anymore (and so the comment below isn't relevant). The purpose of the lock and returned `IDisposable` was to prevent concurrent change of credentials. I don't know if it is still required, but there should be a different name for the method.
@int19h , can you please take a look?